website/docs/docs/contributing/testing-a-new-adapter.md

```diff
@@ -0,0 +1,489 @@
+---
+title: "Testing a new adapter"
+id: "testing-a-new-adapter"
+---
+
+:::info
+
+Previously, we offered a packaged suite of tests for dbt adapter functionality: [`pytest-dbt-adapter`](https://github.com/dbt-labs/dbt-adapter-tests). We are deprecating that suite, in favor of the newer testing framework outlined in this document.
+
+:::
+
+This document has two sections:
+1. ["About the testing framework"](#about-the-testing-framework) describes the standard framework that we maintain for using pytest together with dbt. It includes an example that shows the anatomy of a simple test case.
+2. ["Testing your adapter"](#testing-your-adapter) offers a step-by-step guide for using our out-of-the-box suite of "basic" tests, which will validate that your adapter meets a baseline of dbt functionality.
+
+## Prerequisites
+- Your adapter must be compatible with dbt-core **v1.1** or newer
+- You should be familiar with **pytest**: https://docs.pytest.org/
+
+## About the testing framework
+
+dbt-core offers a standard framework for running pre-built functional tests, and for defining your own tests. The core testing framework is built using `pytest`, a mature and standard library for testing Python projects.
+
+The **[`tests` module](https://github.com/dbt-labs/dbt-core/tree/HEAD/core/dbt/tests)** within `dbt-core` includes basic utilities for setting up pytest + dbt. These are used by all "pre-built" functional tests, and make it possible to quickly write your own tests.
+
+Those utilities allow you to do three basic things:
+1. **Quickly set up a dbt "project."** Define project resources via methods such as `models()` and `seeds()`. Use `project_config_update()` to pass configurations into `dbt_project.yml`.
```

Comment:
Just synced with @VersusFacit on something related to starting using this testing framework. It would be great if we can highlight two things: 
- the concept of pytest fixtures and fixture scope is important when writing tests
- we choose the `class` scope for most of our fixtures.